server/utils/passport/local-login.js

```diff
@@ -20,28 +22,24 @@ export default (req, res, next, user) => {
   })
   //If the user exists, serialize the user to the session
   req.login(user, function (err) {
+    if (err) return res.redirect(`${routes.login}?e=${err}`)
+
     const { appDb } = req.app.locals
     const uid = user.uid
 
     appDb.collection(collections.configs).findOne(
       {
         owner: uid,
       },
-      function (err, configObj) {
-        if (err) {
-          console.error(err.message)
-        }
+      function (queryError, configObj) {
+        if (queryError) return res.redirect(`${routes.login}?e=${queryError}`)
+
         if (!configObj) {
-          const defaultConfig = {
-            owner: uid,
-            config: defaultUserConfig,
-            name: 'default',
-            type: 'matrix',
-            readers: [uid],
-            created: new Date().toUTCString(),
-          }
-          appDb.collection(collections.configs).insertOne(defaultConfig)
+          const config = ConfigModel.default({ owner: uid, readers: [uid] })
+
+          appDb.collection(collections.configs).insertOne(config)
```

Comment:
you could make this part of the config model if you want

```js
const ConfigModel = {
  withDefaults: (overrides = {}) => ({
    config: defaultUserConfig,
    name: 'default',
    type: 'matrix',
    created: new Date().toUTCString(),
    ...overrides,
  }),
  save: async (db, configAttributes) => {
    const config = ConfigModel.withDefaults({ owner: uid, readers: [uid] })

    return await db.collection(collections.configs).insertOne(config)
  }
}
```

```suggestion
          const config = await ConfigModel.save(appDb, { owner: uid, readers: [uid] })
```

Reply:
[88c0683](https://github.com/AMP-SCZ/dpdash/pull/293/commits/88c0683e88c4bba873baf2a78d32f796c0c889a5)

Reply:
the method was added to the config model but we still need to use it here.

Reply:
[a13e560](https://github.com/AMP-SCZ/dpdash/pull/293/commits/a13e5607638067485e759753bc273f9ccf668355)